src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs

```diff
@@ -196,6 +196,27 @@ public void AddModelError([NotNull] string key, [NotNull] Exception exception)
         /// </returns>
         public bool TryAddModelError([NotNull] string key, [NotNull] Exception exception)
         {
+
+            if (exception is FormatException)
+            {
+                // Convert FormatExceptions to Invalid value messages.
+                ModelState modelState;
+                TryGetValue(key, out modelState);
+                string errorMessage;
+                if (modelState == null)
```

Comment:
Please add a unit test for each branch of this `if`. Also for adding an exception that isn't a `FormatException` make sure there's no message (this might already be tested)